### DISCUSSION TITLE
Can't Get Lookup to work.

### DISCUSSION THREAD

**Discussion by pharry22** [Dedac/RestDataMappingPicklist#86] [Q&A]
I have created a webservice to return this data.
`{
"page": 1,
"per_page": 2,
"total": 2,
"data": [
{
"id": 7,
"email": "michael.lawson@reqres.in",
"first_name": "Michael",
"last_name": "Lawson",
},
{
"id": 8,
"email": "lindsay.ferguson@reqres.in",
"first_name": "Lindsay",
"last_name": "Ferguson",
}
]
}`
 I have configured a custom control with these parameters. I am basically following the example in the help text. However, when I edit the work item I do not get a picklist of items.

The "SelectedUser" field in DevOps is defined as Text Single Line.

I know my API is being called as it hits a breakpoint but the picklist in the UI is not being populated.

The 4th screenshot shows what I am seeing in DevOps.  Any suggestions as to what I am doing wrong?

![image](https://user-images.githubusercontent.com/13991563/175462175-b653ce0d-9b5c-4367-96ea-4fa04ebb83b6.png)
![image](https://user-images.githubusercontent.com/13991563/175462250-cfd4f66d-8516-4d64-8426-50806d2efee4.png)
![image](https://user-images.githubusercontent.com/13991563/175462309-52d46205-e7e7-4635-80f5-cd8e96b735fa.png)

![image](https://user-images.githubusercontent.com/13991563/175462100-39f856db-0426-43ca-a92b-e9f0fd2c2061.png)

**Comment by Dedac** [comment]
I have a couple suggestions - The first is that I don't see an array named "data" in your example json, though you may have just left it out.   Secondly, check the browser console and network tab in the developer tools to see the result from the rest call and any errors that might be displayed.

**Reply by pharry22** [reply]
Thanks, that has helped I was getting a CORS error which I have resolved and I am now getting data back to the client. But I am getting this error.
![image](https://user-images.githubusercontent.com/13991563/176604502-75f300f1-884f-4bf0-b80b-81e44b10e0c8.png)

I assume because I do not have the returned data in the right format but can't figure out where it is wrong.

The returned data is:

{
"data": [
{
"id": 7,
"email": "michael.lawson@reqres.in",
"first_name": "Michael",
"last_name": "Lawson",
},
{
"id": 8,
"email": "lindsay.ferguson@reqres.in",
"first_name": "Lindsay",
"last_name": "Ferguson",
}
]
}

Which is copied from the ArrayData field in the RestServiceData.ts routine.

The BaseGet function always returns Undefined.

**Comment by Dedac** [comment]
I can't say that I will be able to debug the issue for you - double check that the configuration is correct (key field matches the json field name), you can also add break points inside "RestServiceData" where arrayData is extracted from the response to see if the JS object is correct.  Then you can check the returned value from the fieldSet and see if it is also a valid array.
